core/src/main/java/org/hisp/dhis/android/core/validation/ValidationRule.java

```diff
@@ -52,57 +49,42 @@
 import java.util.List;
 
 @AutoValue
-@JsonDeserialize(builder = $$AutoValue_ValidationRule.Builder.class)
 public abstract class ValidationRule extends BaseNameableObject implements CoreObject {
 
     @Nullable
-    @JsonProperty()
     public abstract String instruction();
 
-    @JsonProperty()
     @ColumnAdapter(ValidationRuleImportanceColumnAdapter.class)
     public abstract ValidationRuleImportance importance();
 
-    @JsonProperty()
     @ColumnAdapter(ValidationRuleOperatorColumnAdapter.class)
     public abstract ValidationRuleOperator operator();
 
-    @JsonProperty()
     @ColumnAdapter(PeriodTypeColumnAdapter.class)
     public abstract PeriodType periodType();
 
-    @JsonProperty()
     public abstract Boolean skipFormValidation();
 
-    @JsonProperty()
     @ColumnAdapter(LeftValidationRuleExpressionColumnAdapter.class)
     public abstract ValidationRuleExpression leftSide();
 
-    @JsonProperty()
     abstract String leftSideExpression();
```

Comment:
These auxiliary properties are suspicious... They are assigned a value in the builder as if both `leftSide()` and `leftSideExpression()` might or should have a value. It could happen that these are the legacy properties that were replaced by `leftSide()` (in this case, leftSide should be nullable in the DTO or have two differents DTO). Maybe we can look at the commit history to understand this and clean it up